src/NuGet.Core/NuGet.CommandLine.XPlat/PublicAPI.Shipped.txt

```diff
@@ -0,0 +1,82 @@
+NuGet.CommandLine.XPlat.AddPackageReferenceCommand
```

Comment:
Should NuGet.CommandLine.xplat be excluded? 

It's used as an exe, https://github.com/NuGet/NuGet.Client/blob/28f5e5c139daf85a4e3530a0bbd1f8c2cc1f9c94/src/NuGet.Core/NuGet.CommandLine.XPlat/NuGet.CommandLine.XPlat.csproj#L14

Reply:
nuget.exe's nupkg contains nuget.exe in the `tools\` folder, so projects can't use `NuGet.CommandLine` as a PackageReference. However, `NuGet.CommandLine.XPlat` is packed like a classlib. Even though its OutputType is exe, it still goes into `lib/tfm/`, meaning projects can not only take a dependency on it, but also call public APIs directly.

For this reason, I say keep it protected by the analyser. Since it's an exe, and not referenced by other projects, it's zero cost to make all new APIs `internal`, rather than `public`.

Reply:
It's not about new APIs, it's about the old ones, specifically see this thread: https://github.com/NuGet/NuGet.Client/pull/3480#discussion_r454718106

The package in question has a very low download count from GitHub, http://nuget.org/packages/nuget.commandline.xplat, so I don't think there's too many people we can break.

While I understand the sentiment, this will lead to us putting unnecessary restrictions on ourselves. 

If we want to be strict about what we consider a library, we should also define what isn't a library, and imo, this isn't a library.

If we are worried we can change the structure of the nupkg.

Reply:
How are we restricted if we make all new APIs in this project `internal`?

I agree we shouldn't publish this nupkg, but this is how we insert into the .NET SDK for now, so it would be a non-trivial change.

I'm not really opposed to exclude this project from the public apis analyzer. But I feel like the restriction to only use `internal` for new APIs is so low impact I'm not yet convinced it would really burden us in any meaningful way if we do use the analyzer on this project.

Reply:
In the PR comment you linked, I wrote something along the line of there being 3 options:

1. Change all APIs to internal. Our `PublicAPI.*.txt` files will always be empty, and it will be enforced by the analyzer (and PR reviewers to question if the PR author adds APIs to the text files). All new APIs, and changes to existing APIs (like constructors in this case) should remain internal.  This breaks everyone using our packages, but it breaks them once.

2. Pretend like it's not published as a nupkg to nuget.org.  Break APIs as you would with any other console or GUI app. This potentially breaks customers using our packages multiple times, and if they complain, we have to explain.

3. Treat this project like any other shipping classlib, even though it's an exe.

I'm leaning towards 1, but I'm scared that maybe the dotnet cli might have at least one codepath where they call us directly, rather than invoking us as a new process, in which case we break on insertion.

Reply:
I just don't want 3. 

We are writing an exe, so I don't want the code to have added complexity because a refactoring had to preserve some unnecessary backward compat.

Reply:
I checked the .NET CLI source, and they never reference our assembly directly. They only invoke us as a new process. This means the only place we intend this package to be used doesn't have any API or ABI dependencies on us.

Therefore, in a separate PR (so it's easy to revert in case there's a problem), I'm going to change everything `public` to `internal`. This PR will keep the project opted in to using the analyzer, but the text files will be empty. The docs I write will contain a brief mention about why, and inform contributors not to make APIs public in this assembly (and therefore not to add APIs into the text files), and code reviewers should not accept a PR that changes NuGet.CommandLine.XPlat's PublicAPI.*.txt files.